### DISCUSSION TITLE
Client disks are too large. How make them smaller and save memory?

### DISCUSSION THREAD

**Discussion by temk** [ltsp/ltsp#444] [Q&A] [answered]
I use LTSP to run diskless machines. With default configuration, each client got disk of size 3.7G, but only 4% of it was used. Is it possible to produce smaller disks and save memory?

**Comment by alkisg** [comment, marked as answer]
The LTSP image is NOT loaded in RAM.
Suppose that you have a 20 GB **local** OS, and you boot a PC with 2 GB RAM with it. Is all the disk loaded in RAM? No. Only the required sectors/clusters are fetched into the RAM buffers, and after a while, the old buffers are discarded.

It's the same in a network disk as well. You can boot an LTSP client with 1 GB RAM while using a 20 GB LTSP image; it won't waste any additional RAM because it's big. The OS will do the same buffering without caring about the disk size.


I'll take the chance to describe a **different** issue about big image sizes though. Suppose that someone has a folder named /movie-collection or /opt/movie-collection or /home2/movie-collection. If he runs `ltsp image /`, then that folder will be copied to the client image, making it unnecessarily larger. To omit that directory from the image, run `man ltsp.conf` and read about `ADD_IMAGE_EXCLUDES`.
Good places for "extra data" that are not included in the LTSP image by default, are /home, /srv and /media.

To discover if such a thing happened to you, you may run:

```shell
sudo mount -o loop,ro /srv/ltsp/images/x86_64.img /mnt
sudo du -sh /mnt/*
sudo umount /mnt
```
Ignore big sizes in /usr or /lib. If you see other big sizes there, you might want to exclude them from the image.

The typical Ubuntu MATE image is around 2.5 GB, so your 3.7 GB sounds OK if you added a few more applications. I just wrote that in case someone else needs it.

**Reply by temk** [reply]
Thank you for prompt answer. I saw root partition of size 4G and worried that it took 4G of memory. Now I see only 500M of RAM used. Thank you for the explanation.
